src/kfactory/kcell.py

```diff
@@ -7656,7 +7662,10 @@ def show(
             else:
                 wtd = repo.working_tree_dir
                 if wtd is not None:
-                    root = Path(wtd) / "build/gds"
+                    if name.startswith("ipython-"):         # IPython Case: use tempfile
+                        root = Path(tempfile.gettempdir()) / "gdsfactory"
```

Comment:
We aren't gdsfactory here ;)

Jokes aside, what do you think of instead using the build-dir (if available) and naming it `ipython.oas` (no lernelid etc)?

Reply:
I don't have a very strong feeling about using the local repo's 'build' directory (as opposed to a temp directory) for saving OAS files generated from IPython calls. My thinking was that 
 (1) oasis files streamed to KLayout from interactive python sessions are likely not meant to be saved, so use a temp directory and let the OS clean them up
 (2) I often have more than one ipython kernel running concurrently (VSCode interactive sessions for different test files) and I wanted to avoid them overwriting each other's Oasis files. Hence tagging the file with some IPython related ID. 

That said I don't think the directory matters much and I don't know if it matters if multiple kernels are overwriting each others' OASIS files once they are already open in KLayout and thus loaded in memory. I would rather conform to whatever conventions work with the rest of the package code. I'll merge in recent changes if necessary and make these changes to the PR the later this morning.

Reply:
Sounds good to me. I also don't have strong opinions on which name to use. The name only has implications on whether klive will open a new layout view or not. I would stick to one  name (as we have now). This ensures that even though the files are not really ephemeral they are at least somewhat limited (i.e. shell gets (ideally) one file, ipython gets one and otherwise one per python file with `__main__`)